dozer-ingestion/src/connectors/delta_lake/reader.rs

```diff
@@ -0,0 +1,100 @@
+use crate::connectors::delta_lake::ConnectorResult;
+use crate::connectors::object_store::map_value_to_dozer_field;
+use crate::connectors::{ColumnInfo, TableInfo};
+use crate::errors::ConnectorError;
+use crate::ingestion::Ingestor;
+use deltalake::datafusion::prelude::SessionContext;
+use dozer_types::ingestion_types::{DeltaLakeConfig, IngestionMessage};
+use dozer_types::types::{Operation, Record, SchemaIdentifier};
+use futures::StreamExt;
+use std::sync::Arc;
+use tokio::runtime::Runtime;
+
+pub struct DeltaLakeReader {
+    config: DeltaLakeConfig,
+}
+
+impl DeltaLakeReader {
+    pub fn new(config: DeltaLakeConfig) -> Self {
+        Self { config }
+    }
+
+    pub fn read(&self, table: &[TableInfo], ingestor: &Ingestor) -> ConnectorResult<()> {
+        for (id, table) in table.iter().enumerate() {
+            Runtime::new()
```

Comment:
Do we need to create new runtime for every run?

Reply:
No need.

Btw I'm curious why we don't make the following interface async:
```rust
pub trait Connector: Send + Sync + Debug {
  
    fn get_schemas(
        &self,
        table_names: Option<Vec<TableInfo>>,
    ) -> Result<Vec<SourceSchema>, ConnectorError>;

    fn start(
        &self,
        last_checkpoint: Option<(u64, u64)>,
        ingestor: &Ingestor,
        tables: Vec<TableInfo>,
    ) -> Result<(), ConnectorError>;
```

Reply:
Async functions are not supported in traits, so we are not using them.

Reply:
Oh, do we consider using nightly version ?

Reply:
Or make `Connector ` async trait?